digits/tools/s3_walker.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import sys
+from boto.s3.connection import S3Connection
+from boto.s3.key import Key
+from boto.s3.prefix import Prefix
+
+
+class S3Walker(object):
+
+    def __init__(self, endpoint, accesskey, secretkey):
+        self.is_secure = endpoint.find('https://') > -1
+        self.host = endpoint[endpoint.index('://')+3:]
+        has_port = self.host.find(':') > -1
+        if has_port:
+            self.port = int(self.host[self.host.index(':')+1:])
+            self.host = self.host[:self.host.index(':')]
+        else:
+            if self.is_secure:
+                self.port = 443
+            else:
+                self.port = 80
+        self.accesskey = accesskey
+        self.secretkey = secretkey
+        self.conn = None
+
+        print('host: ' + self.host)
+        print('is secure: ' + str(self.is_secure))
+        print('port: ' + str(self.port))
```

Comment:
These prints could be replaced with calls to a logger as in https://github.com/NVIDIA/DIGITS/blob/master/digits/tools/parse_folder.py#L462.

Reply:
We removed these and made a couple other modifications to suppress printing to stdout